overlord/devicestate/handlers_install.go

```diff
@@ -1359,24 +1360,41 @@ func createSaveBootstrappedContainer(saveNode string) (secboot.BootstrappedConta
 	return secbootCreateBootstrappedContainer(secboot.DiskUnlockKey(saveEncryptionKey), saveNode), nil
 }
 
-func deleteOldSaveKey(saveMntPnt string) error {
-	// During factory reset createSaveBootstrappedContainerImpl
-	// will save some old keys that need to stay until factory
-	// reset is finished. This is the function that removes those
-	// keys.
-
-	// FIXME: maybe there is better if we had a function returning the devname instead.
+func deleteOldKeys(saveMntPnt string) error {
 	uuid, err := disksDMCryptUUIDFromMountPoint(saveMntPnt)
 	if err != nil {
 		return fmt.Errorf("cannot find save partition: %v", err)
 	}
 
 	diskPath := filepath.Join("/dev/disk/by-uuid", uuid)
 
-	toDelete := map[string]bool{
+	oldPossiblyTPMKeySlots := map[string]bool{
+		"factory-reset-old-fallback": true,
+	}
+
+	oldKey := device.FallbackSaveSealedKeyUnder(boot.InitramfsSeedEncryptionKeyDir)
+
+	// FIXME: we are missing a bit of context here. We should do this only if we are using TPM2.
+	err = secbootRemoveOldCounterHandles(
+		diskPath,
+		oldPossiblyTPMKeySlots,
+		[]string{oldKey},
+	)
+	if err != nil {
+		return fmt.Errorf("could not clean up old counter handles: %v", err)
+	}
+
+	saveFallbackKeyFactory := device.FactoryResetFallbackSaveSealedKeyUnder(boot.InitramfsSeedEncryptionKeyDir)
+	if err := os.Rename(saveFallbackKeyFactory, oldKey); err != nil {
```

Comment:
this probably merits a comment about what is doing, because the name oldKey is  a bit strange in this context as it is actually the target of the rename, so we will still have a oldKey around?

Reply:
I will comment on it. The new key has a temporary name, and the old key has the default name. When it is confirmed to boot, the new key replaces the old key.

Reply:
I have realized there is a potential bug. I will rewrite the code a bit to make clearer.